test/integration/security/access-nested-permissions.js

```diff
@@ -0,0 +1,525 @@
+            actions: ['on', 'get']
+          }
+        }
+      };
+      let testGroup2 = {
+        name: 'TEST GROUP2',
+        permissions: {
+          '/TEST/2/3/5/6': {
+            prohibit: ['on', 'get']
+          }
+        }
+      };
+
+      let testGroup3 = {
+        name: 'TEST GROUP3',
+        permissions: {
+          '/TEST/1/2/3/4': {
+            actions: ['on', 'get']
+          }
+        }
+      };
+
+      addedTestGroup = await serviceInstance.services.security.users.upsertGroup(testGroup, {
+        overwrite: false
+      });
+
+      addedTestGroup2 = await serviceInstance.services.security.users.upsertGroup(testGroup2, {
+        overwrite: false
+      });
+
+      addedTestGroup3 = await serviceInstance.services.security.users.upsertGroup(testGroup3, {
+        overwrite: false
+      });
+
+      const testUser = {
+        username: 'TEST',
+        password: 'TEST PWD',
+        permissions: {}
+      };
+
+      testUser.permissions['/TEMPLATED_ALLOWED/{{user.username}}/8'] = {
+        actions: ['on', 'get']
+      };
+
+      addedTestuser = await serviceInstance.services.security.users.upsertUser(testUser, {
+        overwrite: false
+      });
+
+      await serviceInstance.services.security.users.linkGroup(addedTestGroup, addedTestuser);
+
+      testClient = await serviceInstance.services.session.localClient({
+        username: testUser.username,
+        password: 'TEST PWD'
+      });
+
+      adminClient = await serviceInstance.services.session.localClient({
+        username: '_ADMIN',
+        password: 'happn'
+      });
+    }
+  );
+
+  after('should disconnect', function(callback) {
+    this.timeout(15000);
+
+    if (testClient)
+      testClient.disconnect({
+        reconnect: false
+      });
+    if (adminClient)
+      adminClient.disconnect({
+        reconnect: false
+      });
+
+    setTimeout(function() {
+      serviceInstance.stop(function() {
+        callback();
+      });
+    }, 3000);
+  });
+
+  context('get', function() {
+    it('gets data from an allowed set of nested permissions', async () => {
+      await adminClient.set('/ALLOWED/0', { test: 0 });
+      await adminClient.set('/TEST/1/2/3', { test: 1 });
+      await adminClient.set('/TEST/2/3/4/5/6', { test: 2 });
+      await adminClient.set('/TEST/4/5/6', { test: 3 });
+      await adminClient.set('/TEMPLATED/TEST/1/2', { test: 4 });
+      await adminClient.set('/TEST/5/6', { test: 5 });
+      await adminClient.set('/TEST/5/6/7', { test: 7 });
+      await adminClient.set('/TEST/5/6/7/9', { test: 9 });
+
+      //allowed templated user permission
+      await adminClient.set('/TEMPLATED_ALLOWED/TEST/8', { test: 8 });
+
+      let results = await testClient.get('/ALLOWED/**');
+      test.expect(results[0].test).to.be(0);
+
+      results = await testClient.get('/TEMPLATED_ALLOWED/TEST/8');
+      test.expect(results.test).to.be(8);
+
+      results = await testClient.get('/TEST/**');
```

Comment:
Why is the length 3 - when I see the following allowed permissions for the group:
/TEST/1/2/3
/TEST/2/3/*
/TEST/2/3/4/5
/TEST/5/6
/TEST/5/6/*
/TEST/5/6/7/9
/TEST/7/8
and we set:
/TEST/1/2/3
/TEST/2/3/4/5/6
/TEST/4/5/6
/TEST/5/6
/TEST/5/6/7
/TEST/5/6/7/9

Reply:
2/3/4/5 is prohibited, as is 5/6/*, so from the list above, 4/5/6, 5/6/7 and 5/6/7/9 don't fire.